src/test/testFiles/invalid/semanticErrors/incompatibleArrayConstructor.manimdsl

```diff
@@ -0,0 +1 @@
+let a = Array<Array<number>();
```

Comment:
Is this not a syntax error (missing >)?

Reply:
The grammar just contains `(ARRAY LT)+` and  `GT+`. Not sure how to count the numbers there should be syntactically unless we hardcode it to 2D arrays?

Reply:
I prefer it being semantic because then if they try to make a 3D array (for example), it's a semantic error rather than a syntax error

Reply:
Not sure what everyone else thinks though

Reply:
Can't you just make it ARRAY LT type GT rather than primitive_type. Type checking should probably be the thing at semantic analysis as right now I imagine the error is something like Array<number is an invalid type?